LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationConfigurationService.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.NetworkServer.BasicsStation
+{
+    using System;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using LoRaWan.NetworkServer.BasicsStation.JsonHandlers;
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.Extensions.Logging;
+
+    internal sealed class BasicsStationConfigurationService : IBasicsStationConfigurationService
+    {
+        private const string RouterConfigPropertyName = "routerConfig";
+
+        private readonly LoRaDeviceAPIServiceBase loRaDeviceApiService;
+        private readonly ILoRaDeviceFactory loRaDeviceFactory;
+
+        public BasicsStationConfigurationService(LoRaDeviceAPIServiceBase loRaDeviceApiService,
+                                                 ILoRaDeviceFactory loRaDeviceFactory)
+        {
+            this.loRaDeviceApiService = loRaDeviceApiService;
+            this.loRaDeviceFactory = loRaDeviceFactory;
+        }
+
+        public async Task<string> GetRouterConfigMessageAsync(StationEui stationEui, CancellationToken cancellationToken)
+        {
+            var queryResult = await this.loRaDeviceApiService.SearchByDevEUIAsync(stationEui.ToString());
+            var info = queryResult.Single();
```

Comment:
It will, I also have a test for it (`When_Device_Key_Lookup_Is_Non_Unique_Fails`). `Single` throws if 0 or 2 or more entries are present, as opposed to `First` or `FirstOrDefault`